csrc/quantization/fp8/common.cuh

```diff
@@ -32,25 +61,35 @@ __device__ __forceinline__ float atomicMaxFloat(float* addr, float value) {
   return old;
 }
 
-template <bool is_scale_inverted>
-__device__ __forceinline__ FP8_TYPE scaled_fp8_conversion(float const val,
-                                                          float const scale) {
+template <bool is_scale_inverted, typename fp8_type>
+__device__ __forceinline__ void scaled_fp8_conversion(fp8_type& ret,
+                                                      float const val,
+                                                      float const scale) {
   float x = 0.0f;
   if constexpr (is_scale_inverted) {
     x = val * scale;
   } else {
     x = val / scale;
   }
 
-  float r = fmax(-FP8_E4M3_MAX, fmin(x, FP8_E4M3_MAX));
+  float r = fmax(-FP8_E4M3_ADJUSTED_MAX<fp8_type>::val(),
+                 fmin(x, FP8_E4M3_ADJUSTED_MAX<fp8_type>::val()));
 #ifndef USE_ROCM
-  return static_cast<c10::Float8_e4m3fn>(r);
+  ret = static_cast<fp8_type>(r);
 #else
   // Use hardware cvt instruction for fp8 on rocm
-  return c10::Float8_e4m3fnuz(
-      __hip_cvt_float_to_fp8(r, fp8::fp8_type::__default_saturation,
-                             fp8::fp8_type::__default_interpret),
-      c10::Float8_e4m3fnuz::from_bits());
+  if constexpr (std::is_same_v<fp8_type, c10::Float8_e4m3fn>) {
+    ret =
+        fp8_type(__hip_cvt_float_to_fp8(r, __hip_fp8_e4m3::__default_saturation,
+                                        __hip_fp8_e4m3::__default_interpret),
+                 fp8_type::from_bits());
+  }
+  if constexpr (std::is_same_v<fp8_type, c10::Float8_e4m3fnuz>) {
+    ret = fp8_type(
+        __hip_cvt_float_to_fp8(r, __hip_fp8_e4m3_fnuz::__default_saturation,
+                               __hip_fp8_e4m3_fnuz::__default_interpret),
+        fp8_type::from_bits());
+  }
```

Comment:
Should this be extracted to a template-specialized utility functions for possible future reuse?

Reply:
Done.